test/functional/Java12andUp/src/org/openj9/test/java_lang/Test_Class.java

```diff
@@ -0,0 +1,130 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.test.java_lang;
+
+import java.lang.constant.ClassDesc;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.util.Optional;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+
+/**
+ * This test Java.lang.Class API added in Java 12 and later version.
+ *
+ * new methods:
+ * - describeConstable()
+ * - descriptorString()
+ * - arrayType() 
+ * - componentType(): wrapper for Class.getComponentType(), not tested
+ */
+public class Test_Class {
+    public static Logger logger = Logger.getLogger(Test_Class.class);
+ 
+	/* descriptor strings */
+	String primitiveResult = "I";
+	String objectResult = "Ljava/lang/Object;";
+	String arrayResult = "[D";
+	String arrayResult2 = "[[[D";
+
+	/* test classes */
+	Class<?> primitiveTest = int.class;
+	Class<?> objectTest = new Object().getClass();
+	Class<?> arrayTest = new double[0].getClass();
+	Class<?> arrayTest2 = new double[0][][].getClass();
+
+	/*
+	 * Test Java 12 API Class.descriptorString()
+	 */
+	@Test(groups = { "level.sanity" })
+	public void testClassDescriptorString() {
+		/* primitive type test */
+		logger.debug("testClassDescriptorString: Primitive test result is: " + primitiveTest.descriptorString() + " expected: " + primitiveResult);
+		Assert.assertTrue(primitiveTest.descriptorString().equals(primitiveResult));
```

Comment:
There are a small number of primitives - Z, B, C, S, I, F, D, J, & V.  Seems like we should test them all:

Something like:
```
Object[][] primitiveData = {
	{ boolean.class, "Z" },
	{ byte.class, "B" },
	{ char.class, "C" },
	{ short.class, "S" },
	{ int.class, "I" },
	{ float.class, "F" },
	{ double.class, "D" },
	{ long.class, "J" },
	{ void.class, "V" }
};
for (Object[] params : primitiveData) {
	Assert.assertEquals(((Class<?>)params[0]).descriptorString(), params[1]);
}
```